store/sqlstore/group_supplier.go

```diff
@@ -0,0 +1,781 @@
+		return result
+	}
+
+	if err != nil {
+		result.Err = model.NewAppError("SqlGroupStore.GroupDeleteGroupSyncable", "store.update_error", nil, err.Error(), http.StatusInternalServerError)
+		return result
+	}
+
+	result.Data = groupSyncable
+	return result
+}
+
+// PendingAutoAddTeamMembers returns a slice of UserTeamIDPair that need newly created memberships
+// based on the groups configurations.
+//
+// Typically since will be the last successful group sync time.
+func (s *SqlSupplier) PendingAutoAddTeamMembers(ctx context.Context, since int64, hints ...store.LayeredStoreHint) *store.LayeredStoreSupplierResult {
+	result := store.NewSupplierResult()
+
+	query := `
+		SELECT 
+			GroupMembers.UserId, GroupTeams.TeamId
+		FROM 
+			GroupMembers
+			JOIN GroupTeams 
+			ON GroupTeams.GroupId = GroupMembers.GroupId
+			JOIN UserGroups ON UserGroups.Id = GroupMembers.GroupId
+			JOIN Teams ON Teams.Id = GroupTeams.TeamId
+			%s JOIN TeamMembers 
+			ON 
+				TeamMembers.TeamId = GroupTeams.TeamId 
+				AND TeamMembers.UserId = GroupMembers.UserId
+		WHERE 
+			TeamMembers.UserId IS NULL
+			AND UserGroups.DeleteAt = 0
+			AND GroupTeams.DeleteAt = 0
+			AND GroupTeams.AutoAdd = true
+			AND GroupMembers.DeleteAt = 0
+			AND Teams.DeleteAt = 0
+			AND (GroupMembers.CreateAt >= :Since
+			OR GroupTeams.UpdateAt >= :Since)`
+
+	sql := fmt.Sprintf("%s UNION %s", fmt.Sprintf(query, "LEFT"), fmt.Sprintf(query, "RIGHT"))
+
+	var userTeamIDs []*model.UserTeamIDPair
+
+	_, err := s.GetMaster().Select(&userTeamIDs, sql, map[string]interface{}{"Since": since})
+	if err != nil {
+		result.Err = model.NewAppError("SqlGroupStore.PendingAutoAddTeamMembers", "store.select_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+
+	result.Data = userTeamIDs
+
+	return result
+}
+
+// PendingAutoAddChannelMembers returns a slice of UserChannelIDPair that need newly created memberships
+// based on the groups configurations.
+//
+// Typically since will be the last successful group sync time.
+func (s *SqlSupplier) PendingAutoAddChannelMembers(ctx context.Context, since int64, hints ...store.LayeredStoreHint) *store.LayeredStoreSupplierResult {
+	result := store.NewSupplierResult()
+
+	query := `
+		SELECT 
+			GroupMembers.UserId, GroupChannels.ChannelId
+		FROM 
+			GroupMembers
+			JOIN GroupChannels ON GroupChannels.GroupId = GroupMembers.GroupId
+			JOIN UserGroups ON UserGroups.Id = GroupMembers.GroupId
+			JOIN Channels ON Channels.Id = GroupChannels.ChannelId
+			%s JOIN ChannelMemberHistory 
+			ON 
+				ChannelMemberHistory.ChannelId = GroupChannels.ChannelId 
+				AND ChannelMemberHistory.UserId = GroupMembers.UserId
+		WHERE
+			ChannelMemberHistory.UserId IS NULL
+			AND ChannelMemberHistory.LeaveTime IS NULL
+			AND UserGroups.DeleteAt = 0
+			AND GroupChannels.DeleteAt = 0
+			AND GroupChannels.AutoAdd = true
+			AND GroupMembers.DeleteAt = 0
+			AND Channels.DeleteAt = 0
+			AND (GroupMembers.CreateAt >= :Since
```

Comment:
From original comment: should we test `UpdateAt` for `GroupMembers`? It's not immediately clear why we'd want `CreateAt` semantics for `GroupMembers` and `UpdateAt` semantics for `GroupChannels`.

Reply:
The `GroupMembers` table does not have an `UpdatedAt` field.